### ISSUE TITLE
OpenADAS.wavelength() method ignores isotope's repository

### ISSUE THREAD

**Issue by vsnever** [cherab/core#295]
The `wavelength()` method in OpenADAS class returns the wavelengths for the element ignoring the isotopes:
https://github.com/cherab/core/blob/6a7c9e581ea9dd6f60b8312a16a2324eec446a6e/cherab/openadas/openadas.py#L55-L57
Thus, this method returns identical wavelength for different hydrogen isotopes ignoring deuterium and tritium repositories.

The simplest solution would be to remove the check:
```python
    if isinstance(ion, Isotope):
        ion = ion.element
```
and access the isotope's repository. However, by default we have such repositories only for hydrogen isotopes.

I suggest to continue using the element's repository if the isotope's repository is not available:
```python
    def wavelength(self, ion, charge, transition):
        """
        :param ion: Element object defining the ion type.
        :param charge: Charge state of the ion.
        :param transition: Tuple containing (initial level, final level)
        :return: Wavelength in nanometers.
        """
        if isinstance(ion, Isotope):
            try:
                return repository.get_wavelength(ion, charge, transition, repository_path=self._data_path)
            except RuntimeError:
                print("Warning! Requested wavelength (isotope={}, charge={}, transition={}) is not available. "
                      "Will try the element's repository.".format(ion.symbol, charge, transition))
                return repository.get_wavelength(ion.element, charge, transition, repository_path=self._data_path)

        return repository.get_wavelength(ion, charge, transition, repository_path=self._data_path)
```
Is the suggested solution ok?

**Comment by Mateasek** [comment]
I just came across this problem! What a chance. I think that anyway for protium-deuterium-tritium the wavelength shift is the strongest, isn't it? I think that the suggested fix is good.

**Comment by CnlPepper** [comment]
I think it would be better to switch to isotope addressing and provide an attribute on the OpenAdas object to enable the element fallback e.g. adas.wavelength_element_fallback = True. By default it should be False, as per the CHERAB rule that anything that can implicitly selects data must be explicitly set by the user.

You'd probably therefore need to define a "hydrogen" wavelength "fallback" lookup table and separate protium, deuterium and tritium lookup tables.

**Comment by vsnever** [comment]
Hi @CnlPepper, thanks, I agree that `wavelength_element_fallback` option is the best solution. 

We already have separate lookup tables for protium, deuterium and tritium. There is no need to create a fallback lookup table for hydrogen, because the lookup table is accessed by the isotope/element symbol and hydrogen and protium share the same symbol.

**Comment by Mateasek** [comment]
Hydrogen isotopes are separated by having a different symbol, which will not be possible for all the other elements. Maybe we should think of how to build such table as @CnlPepper suggested. There is a framework for dealing with isotopes in the core, maybe we could propagate it to the openadas wavelengths too?

**Comment by vsnever** [comment]
> Hydrogen isotopes are separated by having a different symbol, which will not be possible for all the other elements. Maybe we should think of how to build such table as @CnlPepper suggested. There is a framework for dealing with isotopes in the core, maybe we could propagate it to the openadas wavelengths too?

In `cherab.core.atomic.elements` each isotope has it's own symbol, e.g. 'He3' and 'He4' for `helium3` and `helium4` isotopes. If the user creates a separate wavelength repository for helium3, it will be stored in the 'he3' folder. If no 'He3' or 'He4' repository exist, the 'He' repository will be used if `wavelength_element_fallback == True`. Maybe I just don't understand the problem, but it seems to me that everything is ok.

**Comment by Mateasek** [comment]
> In cherab.core.atomic.elements each isotope has it's own symbol, e.g. 'He3' and 'He4' for helium3 and helium4 isotopes. If the user creates a separate wavelength repository for helium3, it will be stored in the 'he3' folder. If no 'He3' or 'He4' repository exist, the 'He' repository will be used if wavelength_element_fallback == True. Maybe I just don't understand the problem, but it seems to me that everything is ok.

You are right, that is sufficient.

**Comment by vsnever** [comment]
Fixed in #296.
